tracer/src/Datadog.Trace/SpanContext.cs

```diff
@@ -111,10 +114,14 @@ internal SpanContext(ulong? traceId, ulong spanId, int? samplingPriority, string
         /// <param name="spanId">The propagated span id.</param>
         /// <param name="rawTraceId">Raw trace id value</param>
         /// <param name="rawSpanId">Raw span id value</param>
-        internal SpanContext(ISpanContext parent, TraceContext traceContext, string serviceName, ulong? traceId = null, ulong? spanId = null, string rawTraceId = null, string rawSpanId = null)
+        internal SpanContext(ISpanContext parent, TraceContext traceContext, string serviceName, TraceId traceId = default, ulong spanId = 0, string rawTraceId = null, string rawSpanId = null)
             : this(parent?.TraceId > 0 ? parent.TraceId : traceId, serviceName)
         {
-            SpanId = spanId ?? RandomIdGenerator.Shared.NextSpanId();
+            // TODO: ^^^ try to get 128-bit parent.TraceId above
```

Comment:
// TODO: ^^^ try to get 128-bit parent.TraceId above if it is a SpanContext

might be a good time to come up with a new internal SpanContext interface and expose the full trace id there? 🧠 🌩️

Reply:
Done by using a helper method.